File: backend/app/database/models/student_model.py
from sqlalchemy import Column, Integer, String, ForeignKey, TIMESTAMP
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database.session import Base

class Student(Base):
    __tablename__ = "students"

    student_id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.user_id", ondelete ="CASCADE"), index=True)
    nim = Column(Integer, unique=True)
    full_name = Column(String(55))
    majors = Column(String(155))
    study_program = Column(String(155))
    created_at = Column(TIMESTAMP, server_default=func.now())
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now())

    # Relationships
    user = relationship("User", back_populates="student", passive_deletes=True)
    submissions = relationship(
        "StudentSubmission",
        back_populates="student",
        cascade="all, delete"
    )

    submission_statuses = relationship(
        "SubmissionStatus",
        back_populates="student",
        cascade="all, delete"
    )

File: frontend/lib/app/data/models/table/student_model.dart
class StudentModel {
  final int? nim;
  final String fullName;
  final String? majors;
  final String? studyProgram;
  final int? studentId;
  final int? userId;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  StudentModel({
    this.nim,
    required this.fullName,
    this.majors,
    this.studyProgram,
    this.studentId,
    this.userId,
    this.createdAt,
    this.updatedAt,
  });

  factory StudentModel.fromJson(Map<String, dynamic> json) => StudentModel(
    nim: json["nim"],
    fullName: json["full_name"] ?? '',
    majors: json["majors"],
    studyProgram: json["study_program"],
    studentId: json["student_id"],
    userId: json["user_id"],
    createdAt: json["created_at"] == null
        ? null
        : DateTime.parse(json["created_at"]),
    updatedAt: json["updated_at"] == null
        ? null
        : DateTime.parse(json["updated_at"]),
  );

  Map<String, dynamic> toJson() => {
    "nim": nim,
    "full_name": fullName,
    "majors": majors,
    "study_program": studyProgram,
    "student_id": studentId,
    "user_id": userId,
    "created_at": createdAt?.toIso8601String(),
    "updated_at": updatedAt?.toIso8601String(),
  };
}
